src/System.Private.CoreLib/src/System/Runtime/InteropServices/NonPortable.cs

```diff
@@ -90,6 +91,18 @@ public static IntPtr GetHINSTANCE(Module m)
                 throw new ArgumentNullException(nameof(m));
             }
 
+
+            RuntimeModule rtModule = m as RuntimeModule;
```

Comment:
This makes the test pass, but the extra logic has dubious value.

Would it make more sense to just leave the test disabled; or even change the Windows path to return `-1` instead of throwing exceptions?

Reply:
I've checked what the docs say and it looks like we should only throw when `m == null` and otherwise return `-1`. So there's definitely an argument for changing the Windows side to return -1.

Reply:
@jkotas I've changed it to return `-1` when `m != null` but `m` is not a runtime module.